lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2913,49 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
+  if (StartsRelativeBoundsClause(Tok)) {
+    ParseRelativeBoundsClause(Result.get());
+  }
+
   return Result;
 }
 
+bool Parser::ParseRelativeBoundsClause(Expr *Expr) {
+  RelativeBoundsClause *Result;
+  ExprResult Results;
```

Comment:
I think you need to replace this with a Boolean variable that indicates an error, instead of using ExprResult.